samcli/lib/build/app_builder.py

```diff
@@ -430,6 +430,11 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             LOG.debug("%s image is built for %s function", build_image, function_name)
         except docker.errors.BuildError as ex:
             LOG.error("Failed building function %s", function_name)
+            for log in ex.build_log:
```

Comment:
Curios Questions: If ex.build_log can be large, iterating over it and writing output in a potentially synchronous manner might be slow. Should we consider case whether `_stream_writer.write_str()` is buffered or should be optimized for larger outputs. I checked here https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/utils/stream_writer.py#L52